da/celestia/celestia.go

```diff
@@ -22,30 +27,31 @@ import (
 
 // DataAvailabilityLayerClient use celestia-node public API.
 type DataAvailabilityLayerClient struct {
-	_      *openrpc.Client
-	client *cnc.Client
+	rpc *openrpc.Client
 
-	namespaceID types.NamespaceID
-	config      Config
-	logger      log.Logger
+	namespace appns.Namespace
+	config    Config
+	logger    log.Logger
 }
 
 var _ da.DataAvailabilityLayerClient = &DataAvailabilityLayerClient{}
 var _ da.BlockRetriever = &DataAvailabilityLayerClient{}
 
+var ErrNotAvailable = errors.New("share: data not available")
+
 // Config stores Celestia DALC configuration parameters.
 type Config struct {
-	BaseURL  string        `json:"base_url"`
-	Timeout  time.Duration `json:"timeout"`
-	Fee      int64         `json:"fee"`
-	GasLimit uint64        `json:"gas_limit"`
+	AuthToken string        `json:"auth_token"`
+	BaseURL   string        `json:"base_url"`
+	Timeout   time.Duration `json:"timeout"`
+	Fee       int64         `json:"fee"`
+	GasLimit  uint64        `json:"gas_limit"`
 }
 
 // Init initializes DataAvailabilityLayerClient instance.
-func (c *DataAvailabilityLayerClient) Init(
-	namespaceID types.NamespaceID, config []byte, kvStore ds.Datastore, logger log.Logger,
-) error {
-	c.namespaceID = namespaceID
+func (c *DataAvailabilityLayerClient) Init(namespaceID types.NamespaceID, config []byte, kvStore ds.Datastore, logger log.Logger) error {
+	// FIXME: prefix 2 zero bytes for compat
```

Comment:
This won't be necessary once https://github.com/celestiaorg/celestia-node/pull/2376 is merged. It provides handy constructor that does all the prepens for you. The constructor is for the type introduced in https://github.com/celestiaorg/celestia-node/pull/2367

Reply:
Using `NamespaceFromBytes`, however `types.NamespaceID` is `[8]byte`, so still need a compatibility patch. Added a TODO to replace `types.NamespaceID` with `share.Namespace`.

Reply:
You can either change your namespace type or use the NewV0Namespace constructor that prepends necessary versions and zeroes for you automatically

Reply:
Added `4bfdaf19300182bc15380a536008c6da4643190a`